Source/DataSources/KmlDataSource.js

```
@@ -1846,6 +2342,221 @@ define([
+        var changed = false;
+        networkLinks.values.forEach(function(networkLink) {
+            var entity = networkLink.entity;
+            if (entitiesToIgnore.contains(entity.id)) {
+                return;
+            }
+
+            if (!networkLink.updating) {
+                var doUpdate = false;
+                if (networkLink.refreshMode === RefreshMode.INTERVAL) {
+                    if (JulianDate.secondsDifference(now, networkLink.lastUpdated) > networkLink.time) {
+                        doUpdate = true;
+                    }
+                }
+                else if (networkLink.refreshMode === RefreshMode.EXPIRE) {
+                    if (JulianDate.greaterThan(now, networkLink.time)) {
+                        doUpdate = true;
+                    }
+
+                } else if (networkLink.refreshMode === RefreshMode.STOP) {
+                    if (cameraViewUpdate) {
+                        networkLink.needsUpdate = true;
+                        networkLink.cameraUpdateTime = now;
+                    }
+
+                    if (networkLink.needsUpdate && JulianDate.secondsDifference(now, networkLink.cameraUpdateTime) >= networkLink.time) {
+                        doUpdate = true;
+                    }
+                }
+
+                if (doUpdate) {
+                    recurseIgnoreEntities(entity);
+                    networkLink.updating = true;
+                    var newEntityCollection = new EntityCollection();
+                    var href = joinUrls(networkLink.href, makeQueryString(networkLink.cookie, networkLink.queryString, true), false);
+                    href = processNetworkLinkQueryString(that._camera, that._canvas, href, networkLink.viewBoundScale, lastCameraView.bbox);
+                    load(that, newEntityCollection, href)
+                        .then(getNetworkLinkUpdateCallback(that, networkLink, newEntityCollection, newNetworkLinks));
```

We probably want an `otherwise` here that raises the `error` event and log the error to the console; otherwise we might end up with some silent failures.